packages/plugin-expo-eas-sourcemaps/src/android.js

```diff
@@ -0,0 +1,36 @@
+const { AndroidConfig, withAndroidManifest } = require('@expo/config-plugins')
+
+// Using helpers keeps error messages unified and helps cut down on XML format changes.
+const { addMetaDataItemToMainApplication, getMainApplicationOrThrow } = AndroidConfig.Manifest
+
+const withAndroidPlugin = (config, onPremConfig) => {
+// Set the Bugsnag API key in the android manifest - note that this isn't required for source map uploads
+// but is added here to support reporting builds using the fastlane plugin
+  config = withAndroidManifest(config, config => {
+    config.modResults = setBugsnagConfig(config, config.modResults)
+    return config
+  })
+
+  return config
+}
+
+// Splitting this function out of the mod makes it easier to test.
+function setBugsnagConfig (config, androidManifest) {
+  const apiKeyName = 'com.bugsnag.android.API_KEY'
+  const apiKeyValue = config?.extra?.bugsnag?.apiKey
+
+  // Get the <application /> tag and assert if it doesn't exist.
+  const mainApplication = getMainApplicationOrThrow(androidManifest)
+
+  addMetaDataItemToMainApplication(
+    mainApplication,
+    apiKeyName,
+    apiKeyValue
+  )
```

Comment:
```suggestion
  if (apiKeyValue) {
    addMetaDataItemToMainApplication(
      mainApplication,
      apiKeyName,
      apiKeyValue
    )
  }
```

Is it okay to set this value if it's undefined?

Reply:
Good point - that's what the code was doing before (and is still the case on versions <48) but it doesn't make much sense to add it if it's undefined